coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/functions/HiveJsonTupleOperator.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2021 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.hive.hive2rel.functions;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlCallBinding;
+import org.apache.calcite.sql.SqlFunctionalOperator;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlOperandCountRange;
+import org.apache.calcite.sql.SqlOperatorBinding;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.type.SqlOperandCountRanges;
+import org.apache.calcite.sql.type.SqlOperandTypeChecker;
+import org.apache.calcite.sql.type.SqlTypeName;
+import org.apache.calcite.sql.validate.SqlValidator;
+import org.apache.calcite.util.Static;
+
+
+public class HiveJsonTupleOperator extends SqlFunctionalOperator {
+  public static final HiveJsonTupleOperator JSON_TUPLE = new HiveJsonTupleOperator();
+
+  public HiveJsonTupleOperator() {
+    super("json_tuple", SqlKind.OTHER_FUNCTION, 200, true, null, null, null);
+  }
+
+  @Override
+  public SqlOperandCountRange getOperandCountRange() {
+    return SqlOperandCountRanges.from(2);
+  }
+
+  @Override
+  protected void checkOperandCount(SqlValidator validator, SqlOperandTypeChecker argType, SqlCall call) {
+    if (call.operandCount() < 2) {
+      throw validator.newValidationError(call, Static.RESOURCE.wrongNumOfArguments());
+    }
+  }
+
+  @Override
+  public boolean checkOperandTypes(SqlCallBinding callBinding, boolean throwOnFailure) {
+    // TODO respect throwOnFailure
+    for (int operand = 0; operand < callBinding.getOperandCount(); operand++) {
+      RelDataType operandType = callBinding.getOperandType(operand);
+      if (!operandType.getSqlTypeName().equals(SqlTypeName.VARCHAR)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  @Override
+  public RelDataType inferReturnType(SqlOperatorBinding opBinding) {
+    RelDataTypeFactory.Builder builder = opBinding.getTypeFactory().builder();
+    for (int i = 0; i < opBinding.getOperandCount() - 1; i++) {
+      builder.add("c" + i, SqlTypeName.VARCHAR);
```

Comment:
c0, c1 are implicit columns names when `json_tuple` is used without explicit aliasing

Reply:
see https://github.com/trinodb/trino/issues/7242#issuecomment-795685037 > "unalias case"

The unaliased case is not supported yet, but still it's better to use that naming as it will make it easier to add unaliased case support in the future.

Reply:
Can we throw an explicit exception specific to the aliasing case?

Reply:
The check is already there:
https://github.com/findepi/coral/blob/7234a991053277068507e697590d93e4eeeb0141/coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/parsetree/ParseTreeBuilder.java#L202